x/ccv/provider/keeper/migration_test.go

```diff
@@ -0,0 +1,193 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/store"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	types2 "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	"github.com/cosmos/interchain-security/v2/testutil/crypto"
+	testutil "github.com/cosmos/interchain-security/v2/testutil/keeper"
+	consumertypes "github.com/cosmos/interchain-security/v2/x/ccv/consumer/types"
+	providerkeeper "github.com/cosmos/interchain-security/v2/x/ccv/provider/keeper"
+	providertypes "github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/v2/x/ccv/types"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmdb "github.com/tendermint/tm-db"
+)
+
+func TestMigrateParamsv1Tov2(t *testing.T) {
+	// Setup raw store
+	db := tmdb.NewMemDB()
+	stateStore := store.NewCommitMultiStore(db)
+	storeKey := sdk.NewKVStoreKey(paramtypes.StoreKey)
+	memStoreKey := storetypes.NewMemoryStoreKey("mem_key")
+	stateStore.MountStoreWithDB(storeKey, sdk.StoreTypeIAVL, db)
+	stateStore.MountStoreWithDB(memStoreKey, sdk.StoreTypeMemory, nil)
+	require.NoError(t, stateStore.LoadLatestVersion())
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
+	ctx := sdk.NewContext(stateStore, tmproto.Header{}, false, log.NewNopLogger())
+	require.NoError(t, stateStore.LoadLatestVersion())
+
+	// Create new empty subspace
+	subspace := paramtypes.NewSubspace(cdc,
+		codec.NewLegacyAmino(),
+		storeKey,
+		memStoreKey,
+		paramtypes.ModuleName,
+	)
+
+	// Note that new param key table is set in keeper constructor
+	subspace = subspace.WithKeyTable(v1KeyTable())
```

Comment:
Wouldn't the func need to be defined with a pointer receiver to update the receiver subspace? See https://github.com/cosmos/cosmos-sdk/blob/d3f09c222243bb3da3464969f0366330dcb977a8/x/params/types/subspace.go#L50

AFAIK value receivers operate on a copy of the original variable

Reply:
Feel free to ignore these small details and merge. But `WithKeyTable()` updates `subspace.table.m`  which is a map type and also a reference type so that's why this func updates the receiver subspace even without using a pointer.

Reply:
This sent me down a quick rabbit hole about the quirks of golang lol, TLDR you are correct 👍 

chatGPTs explanation:

> if a receiver is a struct and its fields are reference types (e.g., maps or slices), modifications made to those fields inside the method will be visible outside the method, even without using a pointer receiver.

Imo it'd still be best practice to just define `WithKeyTable()` with a pointer receiver if it's intended to be modified, instead of relying on a subtle behavior of golang. But it's not our code so I'll let it go :)